kategory-core/src/test/kotlin/kategory/data/MapKWTest.kt

```diff
@@ -27,35 +22,13 @@ class MapKWTest : UnitSpec() {
             eq<MapKW<String, Int>>() shouldNotBe null
         }
 
-        val monoid = MapKW.monoid<String, Int>(SG)
-
-        "Monoid Laws: identity" {
-            val identity = monoid.empty()
-
-            forAll { a: Int ->
-                val map = mapOf("key" to a).k()
-                monoid.combine(identity, map)["key"] == map["key"]
-            }
-
-            forAll { a: Int ->
-                val map = mapOf("key" to a).k()
-                monoid.combine(map, identity)["key"] == map["key"]
-            }
-        }
-
-        "Semigroup laws: associativity" {
-            forAll { a: Int, b: Int, c: Int ->
-                val mapA = mapOf("key" to a).k()
-                val mapB = mapOf("key" to b).k()
-                val mapC = mapOf("key" to c).k()
-
-                monoid.combine(mapA, monoid.combine(mapB, mapC)) == monoid.combine(monoid.combine(mapA, mapB), mapC)
-            }
-        }
-
         testLaws(
+            SemigroupLaws.laws(MapKW.semigroup(), { n -> MapKW(mapOf("key" to n)) }, Eq.any()),
+            MonoidLaws.laws(MapKW.monoid(), { n -> mapOf("key" to n).k() }, Eq.any()),
```

Comment:
This law fails because the result of `combine(MapKW(map={key=value}), M.empty)`  is `MapKW(map=MapKW(map={key=value}))` this occurs because the `combine` method implementation of `MapKWSemigroupInstance` uses a`.k()` inside the method `MapKW.updated`, for this reason the assertion is always false